### PR TITLE
Readable Debug impl for TransactionEffectsDigest

### PR DESCRIPTION
**Pull request by mystenmark**

### CODE REVIEW

File: crates/sui-types/src/base_types.rs [hunk 1]
```diff
@@ -245,7 +245,7 @@ pub struct ObjectDigest(
 
 #[serde_as]
 #[derive(
-    Eq, PartialEq, Ord, PartialOrd, Copy, Clone, Hash, Serialize, Deserialize, JsonSchema, Debug,
+    Eq, PartialEq, Ord, PartialOrd, Copy, Clone, Hash, Serialize, Deserialize, JsonSchema,
```
Comment (kchalkias) [inline crates/sui-types/src/base_types.rs:248]: now it fits in one line

File: crates/sui-types/src/base_types.rs [hunk 2]
```diff
@@ -570,6 +570,14 @@ impl std::fmt::Debug for TransactionDigest {
     }
 }
 
+impl std::fmt::Debug for TransactionEffectsDigest {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::result::Result<(), std::fmt::Error> {
+        let s = base64ct::Base64::encode_string(&self.0);
```
Comment (kchalkias) [inline crates/sui-types/src/base_types.rs:575]: By readable we mean printable String? because previously it was raw (potentially unprintable) bytes?

Comment (bmwill) [reply crates/sui-types/src/base_types.rs:575]: yeah so by default it prints out an array of bytes [54, 32, 0, 1....] which is very difficult to read :)

**Review by bmwill — APPROVED** [review]

File: crates/sui-types/src/base_types.rs (hunk 2 above)
Comment (mystenmark) [reply crates/sui-types/src/base_types.rs:575]: Yes, this is just for parity with the TransactionDigest debug impl

**Pull request merged**